src/components/LmsConfigurations/SuccessFactorsConfigForm.jsx

```diff
@@ -0,0 +1,374 @@
+      return undefined;
+    } catch (error) {
+      return this.handleErrors(error);
+    }
+  }
+
+  updateSuccessFactorsConfig = async (formData, configId) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', this.props.enterpriseId);
+    try {
+      const response = await LmsApiService.updateSuccessFactorsConfig(transformedData, configId);
+      this.setState({ config: response.data });
+      return undefined;
+    } catch (error) {
+      return this.handleErrors(error);
+    }
+  }
+
+  /**
+   * Validates this form. If the form is invalid, it will return the fields
+   * that were invalid. Otherwise, it will return an empty object.
+   * @param {FormData} formData
+   * @param {[String]} requiredFields
+   */
+  validateSuccessFactorsConfigForm = (formData, requiredFields) => {
+    const invalidFields = requiredFields
+      .filter(field => !formData.get(field))
+      .reduce((prevFields, currField) => ({ ...prevFields, [currField]: true }), {});
+    return invalidFields;
+  }
+
+  /**
+   * attempt to submit the form data and show any error states or invalid fields.
+   * @param {FormData} formData
+   */
+  handleSubmit = async (formData, config) => {
+    this.setState({ submitState: SUBMIT_STATES.PENDING, error: null });
+    const requiredFields = [...REQUIRED_SUCCESS_FACTOR_CONFIG_FIELDS];
+
+    // validate the form
+    const invalidFields = this.validateSuccessFactorsConfigForm(formData, requiredFields);
+    if (!isEmpty(invalidFields)) {
+      this.setState({
+        invalidFields: {
+          ...invalidFields,
+        },
+        submitState: SUBMIT_STATES.default,
+      });
+      return;
+    }
+
+    if (config) {
+      const err = await this.updateSuccessFactorsConfig(formData, config.id);
+      if (err) {
+        this.setState({
+          submitState: SUBMIT_STATES.ERROR,
+          error: err,
+        });
+      } else {
+        this.setState({ submitState: SUBMIT_STATES.COMPLETE });
+      }
+    } else {
+      // ...or create a new configuration
+      const err = await this.createSuccessFactorsConfig(formData);
+      if (err) {
+        this.setState({
+          submitState: SUBMIT_STATES.ERROR,
+          error: err,
+        });
+      } else {
+        this.setState({ submitState: SUBMIT_STATES.COMPLETE });
+      }
+    }
+  }
+
+  render() {
+    const {
+      invalidFields,
+      submitState,
+      active,
+      error,
+      transmitTotalHours,
+    } = this.state;
+    const { config } = this.props;
+    let errorAlert;
+    if (error) {
+      errorAlert = (
+        <div className="form-group is-invalid align-items-left">
+          <StatusAlert
+            alertType="danger"
+            iconClassName="fa fa-times-circle"
+            title="Unable to submit config form:"
+            message={error}
+          />
+        </div>
+      );
+    }
+
+    return (
+      <form
```

Comment:
I'm not sure from this PR exactly how the component is used, but if it's basically the whole page, then it needs a wrapper to give it the proper padding / margins etc that the other pages have. (Ignore me if that's already done!)

Reply:
I don't know what you mean full page. There is separate header, sidebar, and footer components that are always there. If there is some margin-ing to do for the sidebar components that I haven't done here, definitely let me know. 

It might be obvious, but most of what I have done is modeled off of the Saml Config forms which was modeled off the ReportingConfig form. I tried to go with consistency when first implementing this stuff how many months ago now? If we have a standard I can look at (or a form that has already been updated to match some standard) I'm more than happy to update these components to match. I just know me and 99.9% of other devs not focused on front end are just gonna copy and paste stuff that seems to work without paying attention to style updates/preferences (I think I recall there being some bug or something because the reporting config didn't match something in the rebrand and thus all these other forms that were based on that also don't match)

Reply:
Sorry the TL;DR is: I don't know what standard I'm missing. If there is 1 (or more) sidebar components that have been updated in some consistent manner, definitely point me at them and I'll make sure this (and all the other LMS configs) match. I just didn't find consistency in how they were implemented 5+ months ago and have been just been basing off of one mostly relevant form (Reporting Config), so they may be currently consistently wrong.

Reply:
No worries. Can you throw a screenshot of this page up? I'm not even 100% sure this is still an issue (I noticed it in the demo, but I've got devstack shut down right now for Paragon work.) 
I'm happy to open it up as a small follow-on ticket if the margin is looking off.

Reply:
Yeah I mean if you saw an issue in the demo, I definitely haven't done anything with extra divs/css or Paragon itself that would fix it lol, but I've got some screenshots here:
![Screen Shot 2020-12-16 at 9 57 07 AM](https://user-images.githubusercontent.com/66267747/102365013-1d514e00-3f85-11eb-8fa8-084857b5ecb8.png)
![Screen Shot 2020-12-16 at 9 57 16 AM](https://user-images.githubusercontent.com/66267747/102365019-1f1b1180-3f85-11eb-9631-b72b1bf71e85.png)

Reply:
Yeah, it needs a bit more margin. I'm gonna open a ticket on it, because it affects both your and @alex-sheehan-edx PRs.

Reply:
Thank you!